eng/Versions.props

```diff
@@ -23,9 +23,6 @@
     <ReportGeneratorVersion>4.3.6</ReportGeneratorVersion>
     <!-- Use the correct compiler version -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
```

Comment:
@jmarolf @mmitche I did not notice this in the original PR, but should we also be removing `UsingToolMicrosoftNetCompilers` or that is required to use the arcade compiler toolset?